napari/layers/layergroup/layergroup.py

```diff
@@ -0,0 +1,183 @@
+from ..base import Layer
+from ..utils.layer_utils import combine_extents
+
+
+class Layergroup(Layer):
+    def __init__(
+        self, children=None, *, name='LayerGroup', ndim=2, visible=True
+    ) -> None:
+        if children is not None:
+            ndim = self._get_ndim()
+        super().__init__(None, ndim)
+        self._name = name
+        if children is None:
+            self._children = []
+        else:
+            self._children = children
+
+    def _render(self):
+        """Recursively return list of strings that can render ascii tree."""
+        lines = []
+        lines.append(self.name)
+
+        for n, child in enumerate(self._children):
+            try:
+                child_tree = child._render()
+            except AttributeError:
+                child_tree = [child.name]
+            lines.append('  +--' + child_tree.pop(0))
+            spacer = '   ' if n == len(self) - 1 else '  |'
+            lines.extend([spacer + l for l in child_tree])
+
+        return lines
+
+    def __str__(self):
+        """Render ascii tree string representation of this layer group"""
+        return "\n".join(self._render())
+
+    def _render_repr(self):
+        """Recursively return list of strings for unambiguous representation"""
+        lines = []
+        cls = type(self)
+        lines.append(f"<{cls.__name__} '{self.name}' at {hex(id(self))}>")
+
+        for n, child in enumerate(self._children):
+            try:
+                child_tree = child._render_repr()
+            except AttributeError:
+                cls = type(child)
+                child_tree = [
+                    f"<{cls.__name__} layer '{child.name}' at "
+                    f"{hex(id(self))}>"
+                ]
+            lines.append('  +--' + child_tree.pop(0))
+            spacer = '   ' if n == len(self) - 1 else '  |'
+            lines.extend([spacer + l for l in child_tree])
+
+        return lines
+
+    def __repr__(self):
+        """Render unambiguous tree string representation of this layer group"""
+        return "\n".join(self._render_repr())
+
+    def __iter__(self):
+        """Iteration yields every non-group layer with a depth first search."""
+        for child in self._children:
+            yield from child.__iter__()
```

Comment:
```suggestion
            yield from child
```

don't think the `__iter__()` is necessary there?